### PR TITLE
Split wren libraries off as libwren and build the real CLI interpreter

### LINKED ISSUE #5712: Wren wrong executable
**Problem description**

In wren update 0.3.0 the CLI/REPL was moved into a [separate repository](https://github.com/wren-lang/wren-cli/). The executable currently included in the termux package is wren_test, this can be built from the main wren repo and can run basic wren scripts but can't access, for example, the io module included in the CLI tool.

I think the executable in the module should be the CLI / REPL tool because it's far more useful standalone than wren_test.

I currently don't have access to a PC but I might be able to open a PR for it in about a week.

### PR DESCRIPTION
**Pull request by buttaface**
This builds fine and fixes #5712, but I'm not sure if the `TERMUX_PKG_REPLACES` is set right. Also, we could make `libwren` a dependency of `wren`, just to keep things the same for people who installed the old `wren`, even though the libraries aren't needed to run the CLI interpreter.

@xeffyr or @Grimler91, could one of you review this?

### CODE REVIEW

File: packages/wren/projects-make-wren_cli.make.patch [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+-OBJECTS += $(OBJDIR)/tcp.o
+-OBJECTS += $(OBJDIR)/thread.o
+-OBJECTS += $(OBJDIR)/threadpool.o
+-OBJECTS += $(OBJDIR)/timer.o
+ OBJECTS += $(OBJDIR)/timer1.o
+-OBJECTS += $(OBJDIR)/tty.o
+-OBJECTS += $(OBJDIR)/udp.o
+-OBJECTS += $(OBJDIR)/uv-common.o
+-OBJECTS += $(OBJDIR)/uv-data-getter-setters.o
+-OBJECTS += $(OBJDIR)/version.o
```
Comment (buttaface) [inline packages/wren/projects-make-wren_cli.make.patch:74]: All these removed objects are from the libuv that this project includes.

**Comment by xeffyr** [comment]
@buttaface `*_BREAKS` `*_REPLACES` should be set only for package providing headers. If headers are provided only by `libwren`, then set these variables in its build.sh and remove from `wren`.

**Comment by buttaface** [comment]
Alright, let me know if it's correct now.

**Comment by xeffyr** [comment]
Edited libwren to specify `wren (<< 0.3.0)` instead of just `wren` as otherwise user won't be able to have `libwren` and `wren` installed at the same time.

Looks fine now.

**Pull request merged**